### ISSUE TITLE
Navigate regions style improvements

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#8553]
When jumping through the main editor sections with the keyboard shortcut `Ctrl + backtick` there are a few things that could be improved:

1
When the posts or sidebar content is very long, scrolling the content or sidebar makes the focus style shape scroll too:

![screen shot 2018-08-01 at 19 28 58](https://user-images.githubusercontent.com/1682452/43687269-782fe658-98d2-11e8-8364-d06105f25d86.jpg)

![screen shot 2018-08-04 at 17 22 59](https://user-images.githubusercontent.com/1682452/43687270-784f1d52-98d2-11e8-8aa0-54fe851d389d.jpg)

This happens because the focus style is set on a CSS pseudo element that scrolls together with the content because, well it's content. Ideally, the focus style should be set as an outline on the landmark regions so it's always visible regardless of scrolling.

2
When focusing the "Open publish panel" region, part of the content at the bottom of the page becomes not visible:

![screen shot 2018-08-05 at 15 25 38](https://user-images.githubusercontent.com/1682452/43687301-fe920866-98d2-11e8-9e56-9374468183fb.png)

this happens because the "Open publish panel" region is relatively positioned and thus still takes space. Also, it would be nice to place the button in the same exact position of the "Skip to the selected block" button:

![screen shot 2018-08-01 at 19 40 32](https://user-images.githubusercontent.com/1682452/43687326-41db0b4a-98d3-11e8-9cb6-ba60a912fea7.png)

3
With Firefox, the "Open publish panel" region stays out of view when it receives focus.

4
Worth noting the current implementation uses `:focus-within`, not supported by IE11 and Edge. This part should be refactored to use just `:focus`.

https://github.com/WordPress/gutenberg/blob/3450273fe7522101e3809b270c80a7b9a3ebcf3a/edit-post/components/layout/style.scss#L127
